### PR TITLE
MINOR: Use exceptions in o.a.k.common if possible and deprecate ZkUtils

### PR DESCRIPTION
**Pull request by ijuma**
Also:
- Remove exceptions in `kafka.common` that are no longer used.
- Keep `kafka.common.KafkaException` as it's still used by `ZkUtils`,
`kafka.admin.AdminClient` and `kafka.security.auth` classes and
we would like to maintain compatibility for now.
- Add deprecated annotation to `kafka.admin.AdminClient`. The scaladoc
stated that the class is deprecated, but the annotation was missing.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/utils/ZkUtils.scala [hunk 1]
```diff
@@ -183,6 +185,8 @@ object ZkUtils {
 /**
  * Legacy class for interacting with ZooKeeper. Whenever possible, ``KafkaZkClient`` should be used instead.
  */
+@deprecated("This is an internal class that is no longer used by Kafka and will be removed in a future release. Please " +
+  "use org.apache.kafka.clients.admin.AdminClient instead.", since = "2.0.0")
```
Comment (omkreddy) [inline core/src/main/scala/kafka/utils/ZkUtils.scala:189]: use  KafkaZkClient?

Comment (ijuma) [reply core/src/main/scala/kafka/utils/ZkUtils.scala:189]: No, `KafkaZkClient` is an internal class too. People should use the `AdminClient` for admin functionality which is the common reason for external users to use this class. Does that clarify it?

Comment (omkreddy) [reply core/src/main/scala/kafka/utils/ZkUtils.scala:189]: Yes, makes sense.  Thanks for the clarification.

**Review by rajinisivaram — APPROVED** [review]
@ijuma Thanks for the PR, LGTM

**Comment by ijuma** [comment]
Test failure is unrelated (one of the Jenkins jobs passed and one had 1 unrelated failure). Merging to trunk and 2.0.

**Pull request merged**